syntax = "proto2";
package fueling.common.record.kinglong.proto.modules;

import "fueling/common/record/kinglong/proto/modules/header.proto";
import "fueling/common/record/kinglong/proto/modules/perception_obstacle.proto";

message Id {
  optional bytes id = 1;
}

message TrajectoryPoint {
    optional double x = 1;  // in meters.
    optional double y = 2;  // in meters.
    optional double z = 3;  // height in meters.
    optional double velocity = 4;  // velocity, in meters / second
    optional double t = 5; // in milliseconds,  relative to the time_stamp of this message (t = t_this_state - t_timestamp)
    optional double heading = 6; // heading in angle
    optional double confidence = 7; // value range: [0.0, 1.0]
}
message Trajectory {
    optional double probability = 1; //probability of this trajectory
    repeated TrajectoryPoint trajectory_point = 2;
    repeated Id lane_id = 3; // ids of lanes trajectory locates on
}
message PredictionObstacle {
    optional PerceptionObstacle perception_obstacle = 1;
    optional double time_stamp = 2; // GPS time in seconds
    optional double predicted_period = 3; // the length of the time for this prediction (e.g. 10s)
    optional double max_history_speed = 6;
    optional double history_speed_heading = 7;
    repeated Trajectory trajectory = 4; // can have multiple trajectories per obstacle
    optional bool need_avoidance = 5;
}
message PredictionObstacles {
    optional Header header = 1; // timestamp is included in header
    repeated PredictionObstacle prediction_obstacle = 2; // make prediction for multiple obstacles
    optional PerceptionErrorCode perception_error_code = 3; // perception error code
    optional WatchArea watch_area = 4; // perception watch area
}
